daml/ContingentClaims/Lifecycle.daml

```diff
@@ -3,83 +3,120 @@
 -- SPDX-License-Identifier: Apache-2.0
 --
 
-{-# LANGUAGE MultiWayIf #-}
 {-# OPTIONS -Wno-deprecations #-} -- To supress 'Monad' warnings
 
 module ContingentClaims.Lifecycle (
-    lifecycle
-  , lifecycle'
+    settle
+  , settle'
   , Result(..)
 ) where
 
 import ContingentClaims.Claim (Claim, Claim(..), ClaimF(..))
 import ContingentClaims.Observation (Observation, eval)
 import ContingentClaims.Util (apoCataM, pruneZeros')
+import Daml.Control.Recursion (project, embed)
 import Daml.Control.Monad.Writer (WriterT, runWriterT)
 import Daml.Control.Monad.State (StateT(..), evalStateT)
 import Daml.Control.Monad.MonadWriter (MonadWriter(..))
 import Daml.Control.Monad.MonadState (MonadState(..), modify)
 import Daml.Control.Monad.Trans
 import Daml.Control.Arrow (Kleisli(..))
+import DA.Action ((>=>))
+import DA.Traversable (sequence)
 import Prelude hiding (sequence, mapA)
 
 type C a = Claim Observation Date a
 type F a = ClaimF Observation Date a
 
 deriving instance Eq a => Eq (C a)
 
--- | Returned from a lifecycle operation.
+-- | Returned from a `settle` operation.
 data Result a = Result with
   remaining : C a
```

Comment:
Consider swapping order of arguments, for consistency with other data types. The functor tends to be on the right-hand-side. (Think how you'd repeatedly fold over this result - on the `remainder`).